lib/sax.js

```diff
@@ -581,11 +593,27 @@ function parseDCC(source, start, domBuilder, errorHandler) {
 						sysid = matchs[3][0];
 					}
 				}
-				var lastMatch = matchs[len - 1];
-				domBuilder.startDTD(name, pubid, sysid);
+
+				// Unquote pubid and sysid if necessary
+				if (pubid && /^(['"]).*\1$/.test(pubid)) {
+					pubid = pubid.substring(1, pubid.length - 1);
+				}
+				if (sysid && /^(['"]).*\1$/.test(sysid)) {
+					sysid = sysid.substring(1, sysid.length - 1);
+				}
```

Comment:
This change seems to break quite some code regarding mixed quotes inside those ids. So I would recommend to not make that change.
I must admit that this is not documented well, so I will push a change to your PR that fixes the documentation part.

See https://github.com/xmldom/xmldom/blob/c4cb17d48feb2fbf1d4ac5ff96c48aa1e8117612/CHANGELOG.md#050

Reply:
> This change seems to break quite some code regarding mixed quotes inside those ids. So I would recommend to not make that change. I must admit that this is not documented well, so I will push a change to your PR that fixes the documentation part.
> 
> See https://github.com/xmldom/xmldom/blob/c4cb17d48feb2fbf1d4ac5ff96c48aa1e8117612/CHANGELOG.md#050

I understand the decision made at that point, even though strictly speaking the quotes are not a part of the indentifier.

Reply:
If you want to follow that path (since you reverted my change to restore the quotes), you have to take care of fixing the tests that preserve the mixed quote cases in a different way.
Otherwise I can not merge this PR.
And I would prefer to not "fix everything in one PR", but instead have short lived PR with a well defined scope.

Reply:
Yes, everything is fine, I misunderstood your commit message.